fapi-client/src/main/scala/com/gu/facia/api/models/collection.scala

```diff
@@ -155,19 +152,21 @@ object Collection extends StrictLogging {
       Response.Async.Right(futureMaybeAtomData)
     }
 
-    def extractValidMediaAtom(mediaAtoms: collection.Seq[Atom], contentId: String): Option[AtomData.Media] = {
-      mediaAtoms.view.map(_.data).collectFirst {
-        case mediaAtom: AtomData.Media if !isExpired(mediaAtom.media) =>
-          Some(mediaAtom)
-        case mediaAtom: AtomData.Media =>
-          logger.info(s"Media atom in content ID: $contentId is expired")
-          None
-      }.flatten
+    def isValidMediaAtom(atom: Atom, id: String): Boolean = {
+      atom.data match {
+        case mediaData: AtomData.Media =>
+          if (!isExpired(mediaData.media)) {
+            true
+          } else {
+            logger.warn(s"Media atom is expired in ${id}")
+            false
+          }
+        case _ =>
+          logger.warn(s"No valid media atom found in ${id}")
+          false
+      }
```

Comment:
We use the same case match for replacement or main media videos so I've extracted it here.